mlos_core/mlos_core/optimizers/README

```diff
@@ -0,0 +1,27 @@
+# Optimizers
+
+This is a directory that contains wrappers for different optimizers to integrate into MLOS.
+This is implemented though child classes for the `BaseOptimizer` class defined in `optimizer.py`.
+
+The main goal of these optimizers is to `suggest` configurations, possibly based on prior trial data to find an optimum based on some objective(s).
+This process is interacted with through `register` and `suggest` interfaces.
+
+The following definitions are useful for understanding the implementation
+
+- `configuration`: a vector representation of a configuration of a system to be evaluated.
+- `score`: the objective(s) associated with a configuration
+- `metadata`: additional information about the evaluation, such as the runtime budget used during evaluation.
+- `context`: additional (static) information about the evaluation used to extend the internal model used for suggesting samples.
```

Comment:
nit: the section on `context` should appear before `metadata` only because that's the order the arguments happen in